src/main/java/org/jboss/tm/ExtendedJBossXATerminator.java

```diff
@@ -91,8 +91,9 @@
      *                      set in the parameter. These constants are defined in javax.transaction.xa.XAResource interface
      * @return an array of Xids to recover
      * @throws XAException An error has occurred. Possible values are XAER_RMERR, XAER_RMFAIL, XAER_INVAL, and XAER_PROTO.
+     * @throws UnsupportedOperationException Operation is not supported in the context. This happens when Narayana runs on JTS.
      */
-    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException;
+    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException, UnsupportedOperationException;
```

Comment:
@ochaloup its because we don't support EJB remoting with JTS so no-one would ever really run that. EJB remoting is a transport (like CORBA is for JTS).

I missed the missing "?" on Mikes question so it looked like a statement, I will check now....